app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/CreateDBTablePageSolution.java

```diff
@@ -0,0 +1,436 @@
+        
+        mappedTableColumns = mapKeys(sourceTable, destTable);
+        int idx = 0;
+        while (idx < sourceTableColumns.size() && idx < destTableColumns.size()) {
+            
+            if (!mappedTableColumns.containsKey(sourceTableColumns.get(idx).getName())) {
+                mappedTableColumns.put(sourceTableColumns.get(idx).getName(), destTableColumns.get(idx).getName());
+            }
+            idx++;
+        }
+        
+        if (idx < sourceTableColumns.size()) {
+            while (idx < sourceTableColumns.size()) {
+                //This will act as a ref to delete the unwanted fields from actions and layout
+                mappedTableColumns.put(sourceTableColumns.get(idx).getName(), DELETE_FIELD);
+                idx++;
+            }
+        }
+        return mappedTableColumns;
+    }
+    
+    private Map<String, String> mapKeys(Table sourceTable, Table destTable) {
+        Map<String, String> primaryKeyNameMap = new HashMap<>();
+        //keyType vs keyName
+        List<String> sourceKeys = new ArrayList<>();
+        List<String> destKeys = new ArrayList<>();
+        
+        if (sourceTable.getKeys() == null || sourceTable.getKeys().isEmpty()
+            || destTable.getKeys() == null || destTable.getKeys().isEmpty()) {
+            return primaryKeyNameMap;
+        }
+        sourceTable.getKeys().forEach(key -> {
+            if (key instanceof PrimaryKey) {
+                PrimaryKey pKey = (PrimaryKey) key;
+                sourceKeys.addAll(pKey.getColumnNames());
+            }
+        });
+        destTable.getKeys().forEach(key -> {
+            if (key instanceof PrimaryKey) {
+                PrimaryKey pKey = (PrimaryKey) key;
+                destKeys.addAll(pKey.getColumnNames());
+            }
+        });
+        return Streams.zip(sourceKeys.stream(), destKeys.stream(), Maps::immutableEntry)
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    private JSONObject extractAndUpdateAllWidgetFromDSL(JSONObject dsl,
+                                                        Map<String, String> mappedColumnsAndTableNames,
+                                                        Set<String> deletedWidgets) {
+
+        if (dsl.get(FieldName.WIDGET_NAME) == null) {
+            // This isn't a valid widget configuration. No need to traverse this.
+            return dsl;
+        }
+
+        updateTemplateWidgets(dsl, mappedColumnsAndTableNames);
+
+        // Updates in dynamicBindingPathlist not required as it updates on the fly by FE code
+        // Fetch the children of the current node in the DSL and recursively iterate over them
+        ArrayList<Object> children = (ArrayList<Object>) dsl.get(FieldName.CHILDREN);
+        ArrayList<Object> newChildren = new ArrayList<>();
+        if (children != null) {
+            for (Object obj : children) {
+                Map data = (Map) obj;
```

Comment:
Along with the `isEmpty` check we are doing just below, can we do an `obj instanceof Map` check as well?

Reply:
I didn't get you here. Do you mean like this
```
for (Object obj : children) {
                Map data = (Map) obj;
                JSONObject object = new JSONObject();
                // If the children tag exists and there are entries within it
                if (!CollectionUtils.isEmpty(data) && obj instanceof Map) {
                    object.putAll(data);


```
If yes wouldn't be the obj will always be the instanceof Map only at that line, which makes it true always.

Reply:
I was thinking more about checking *before* the typecast, like:

```suggestion
                if (!(obj instanceof Map)) {
                    log.error("Non-map item in DSL Children");
                    continue;
                }
                Map data = (Map) obj;
```